application/src/main/java/org/togetherjava/tjbot/commands/tags/TagCommand.java

```diff
@@ -33,19 +35,22 @@ public TagCommand(TagSystem tagSystem) {
 
         // TODO Thing about adding an ephemeral selection menu with pagination support
         // if the user calls this without id or similar
-        getData().addOption(OptionType.STRING, ID_OPTION, "the id of the tag to display", true);
+        getData().addOption(OptionType.STRING, ID_OPTION, "the id of the tag to display", true)
+            .addOption(OptionType.USER, MENTION_USER, "replying to the user ", false);
     }
 
     @Override
     public void onSlashCommand(@NotNull SlashCommandEvent event) {
         String id = Objects.requireNonNull(event.getOption(ID_OPTION)).getAsString();
+        User mentionUser = Objects.requireNonNull(event.getOption(MENTION_USER)).getAsUser();
```

Comment:
There still is one issue, the option is optional.

And this part of the code removes the optional part, if it's not set it'll throw an error and the command doesn't run

Reply:
* Basically, if its optional, it may return `null`. So `requireNonNull` is wrong. You have to consider the fact that it can be null and then basically skip the mentioning part (with a simple if-else).
* That said, to streamline with my other proposal, rename to `replyToUser` or `userToReplyTo`.